std/conv.d

```diff
@@ -4517,25 +4508,98 @@ if (is(T == class))
         static assert(is(Args[0] : typeof(T.outer)),
             "The first argument must be a pointer to the outer class");
 
-        result.outer = args[0];
+        chunk.outer = args[0];
         alias args1 = args[1..$];
     }
     else alias args1 = args;
 
     // Call the ctor if any
-    static if (is(typeof(result.__ctor(args1))))
+    static if (is(typeof(chunk.__ctor(args1))))
     {
         // T defines a genuine constructor accepting args
         // Go the classic route: write .init first, then call ctor
-        result.__ctor(args1);
+        chunk.__ctor(args1);
     }
     else
     {
         static assert(args1.length == 0 && !is(typeof(&T.__ctor)),
             "Don't know how to initialize an object of type "
             ~ T.stringof ~ " with arguments " ~ typeof(args1).stringof);
     }
-    return result;
+    return chunk;
+}
+
+///
+@safe unittest
+{
+    class SafeClass
+    {
+        int x;
+        @safe this(int x) { this.x = x; }
+    }
+
+    auto buf = new void[__traits(classInstanceSize, SafeClass)];
+    auto support = (() @trusted => cast(SafeClass)(buf.ptr))();
+    auto safeClass = emplace!SafeClass(support, 5);
+    assert(safeClass.x == 5);
+
+    class UnsafeClass
+    {
+        int x;
+        @system this(int x) { this.x = x; }
+    }
+
+    auto buf2 = new void[__traits(classInstanceSize, UnsafeClass)];
+    auto support2 = (() @trusted => cast(UnsafeClass)(buf2.ptr))();
+    static assert(!__traits(compiles, emplace!UnsafeClass(support2, 5)));
```

Comment:
@wilzbach @CyberShadow 
Hello guys!

I'm giving you this ping as I believe we have a bug in the style checker.
The files that get generated remove the `@safe` from the unittests which cause these static asserts to fail.

Could you please have a look at this? Is this behaviour by design? What is your input on this?

Cheers